airbyte-webapp/src/components/hooks/services/Health/HealthPollService.tsx

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState } from "react";
+import { useIntl } from "react-intl";
+
+import { useNotificationService } from "components/hooks/services/Notification/NotificationService";
+import { HealthService } from "core/health/HealthService";
+
+const healthService = new HealthService();
+
+const HEALTH_NOTIFICATION_ID = "health.error";
+
+function useApiHealthPoll(pollPeriod: number): void {
+  const [count, setCount] = useState(0);
+  const { formatMessage } = useIntl();
+  const {
+    registerNotification,
+    unregisterNotificationById,
+  } = useNotificationService();
+
+  useEffect(() => {
+    const errorNotification = {
+      id: HEALTH_NOTIFICATION_ID,
+      title: formatMessage({ id: "notifications.error.health" }),
+      isError: true,
+    };
+
+    const interval = setInterval(async () => {
+      try {
+        await healthService.health();
+        if (count >= 3) {
```

Comment:
Yeah, we do not need to unregister, because it won't be there yet.

Good idea to change it into constant: should it be part of config or just const above is enough? FYI interval is part of config.